### PR TITLE
Bug 1331428 - Use EarlGrey for grey box tests in UITests 

### PR DESCRIPTION
**Pull request by npark-mozilla**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
ButtonToast.swift | 18 | Colons should be next to the identifier when specifying a type. 
EarlGrey.swift | 71 | Colons should be next to the identifier when specifying a type. 
EarlGrey.swift | 77 | Colons should be next to the identifier when specifying a type. 
#### Errors

File | Line | Reason |
| --- | ----- | ----- |
ButtonToast.swift | 23 | Opening braces should be preceded by a single space and on the same line as the declaration. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: EarlGreyTests/EarlGrey.swift [hunk 1]
```diff
@@ -0,0 +1,118 @@
+//
```
Comment (farhanpatel) [inline EarlGreyTests/EarlGrey.swift]: Is it cool if we move these to ThirdParty/EarlGreyTests or UITests/EarlGreyTests. Too many top level folders lol.

**Comment by farhanpatel** [comment]
neat. My only concern is that are we arent adding the Framework to the non-test targets correct? So when we build for Production EarlGrey wont be included right?

I just messed around with Kif yesterday. And one of the issues I found with is is that it failed silently when I was running a test. How does EarlGrey handle a failure if It can't find an element?

File: EarlGreyTests/EarlGrey.swift (hunk 1 above)
Comment (npark-mozilla) [inline EarlGreyTests/EarlGrey.swift]: ah, it was a dead file that i failed to notice, will remove it

**Comment by npark-mozilla** [comment]
So when I was running it on xcode IDE, the earl grey log pinpoints which element were failed to show up (or whether there are multiple element match) so I used that to debug my tests, I didn't see kif generating the same level of detail on its log.   

The UITest only builds on Nightly scheme, and UITest is the one that copies the framework, so I believe EarlGrey won't be added to the production environment.  I'm new to xcode as well, so we should keep an eye on it though.

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
AuthenticationTests.swift | 86 | There should be no space before and one after any comma. 
AuthenticationTests.swift | 106 | There should be no space before and one after any comma. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
AuthenticationTests.swift | 86 | There should be no space before and one after any comma. 
AuthenticationTests.swift | 106 | There should be no space before and one after any comma. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Errors

File | Line | Reason |
| --- | ----- | ----- |
AuthenticationTests.swift | 86 | There should be no space before and one after any comma. 
AuthenticationTests.swift | 106 | There should be no space before and one after any comma. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**